src/pybind/mgr/dashboard/controllers/home.py

```diff
@@ -58,7 +58,7 @@ def __init__(self):
 class HomeController(BaseController, LanguageMixin):
     LANG_TAG_SEQ_RE = re.compile(r'\s*([^,]+)\s*,?\s*')
     LANG_TAG_RE = re.compile(
-        r'^(?P<locale>[a-zA-Z]{1,8}(-[a-zA-Z0-9]{1,8})?)(;q=(?P<weight>[01]\.\d{0,3}))?$')
+        r'^(?P<locale>[a-zA-Z]{1,8}(-[a-zA-Z0-9]{1,8})*|\*)(;q=(?P<weight>[01]\.\d{0,3}))?$')
```

Comment:
Thanks for fixing this one! However according to the RFC you posted, the following changes are needed too (since `*`, `*-*`, `zh-*`, `zh-Hans-*` are allowed too):
```suggestion
        r'^(?P<locale>([a-zA-Z]{1,8}|\*)(-([a-zA-Z0-9]{1,8}|\*))*)(;q=(?P<weight>[01]\.\d{0,3}))?$')
```

Do you mind adding the examples above to the unit test you suggested? Thanks!

Reply:
I don't think so. `zh-*` is allowed in section 2.2 Extended Language Range. But Accept-Language header only allows 2.1 Basic Language Range (https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.5).

Reply:
You're right... but I couldn't find any conclusive answer to which HTTP header the Extended Language Range can be used in (if not in the `Accept-Language`)... Anyway, let's then stick to the Basic Range.